webapp/channels/src/components/post_view/post_list_virtualized/post_list_virtualized.tsx

```diff
@@ -458,6 +462,15 @@ export default class PostList extends React.PureComponent<Props, State> {
                 showSearchHint: offsetFromBottom > this.showSearchHintThreshold,
             });
         }
+
+        if (this.state.isScrollToBottomDismissed) {
+            this.resetIsScrollToBottomToastDismissIfNeccessary();
+        } else {
+            // Not to show scroll-to-bottom toast if the user dismissed it and kept scrolling up.
+            this.setState(({
+                showScrollToBottomToast: offsetFromBottom > this.showSearchHintThreshold, // Initially, follow the threshold of the search hint toast.
+            }));
+        }
```

Comment:
Can you explain this logic? I am having a hard time wrapping my head around it, and probably an explanation can help me visualize it :smile:

Reply:
OMG, it's properly unclear. I'll handle it.